src/livestatus.c

```diff
@@ -0,0 +1,479 @@
+
+livestatus_status_t c_to_livestatus_status(const char **fields) {
+  livestatus_status_t status = {
+      .cached_log_messages = atoi(fields[0]),
+      .connections = strtoull(fields[1], NULL, 10),
+      .connections_rate = strtod(fields[2], NULL),
+      .forks = strtoull(fields[3], NULL, 10),
+      .forks_rate = strtod(fields[4], NULL),
+      .host_checks = strtoull(fields[5], NULL, 10),
+      .host_checks_rate = strtod(fields[6], NULL),
+      .livecheck_overflows = strtoull(fields[7], NULL, 10),
+      .livecheck_overflows_rate = strtod(fields[8], NULL),
+      .livechecks = strtoull(fields[9], NULL, 10),
+      .livechecks_rate = strtod(fields[10], NULL),
+      .log_messages = strtoull(fields[11], NULL, 10),
+      .log_messages_rate = strtod(fields[12], NULL),
+      .neb_callbacks = strtoull(fields[13], NULL, 10),
+      .neb_callbacks_rate = strtod(fields[14], NULL),
+      .requests = strtoull(fields[15], NULL, 10),
+      .requests_rate = strtod(fields[16], NULL),
+      .service_checks = strtoull(fields[17], NULL, 10),
+      .service_checks_rate = strtod(fields[18], NULL),
+  };
+
+  return status;
+} /* livestatus_status_t c_to_livestatus_status */
+
+// Borrowed from https://stackoverflow.com/a/29378380
+static long ls_strtol_subrange(const char *s, char **endptr, int base, long min,
+                               long max) {
+  long y = strtol(s, endptr, base);
+  if (y > max) {
+    errno = ERANGE;
+    return max;
+  }
+  if (y < min) {
+    errno = ERANGE;
+    return min;
+  }
+  return y;
+} /* static long ls_strtol_subrange */
+
+// Borrowed from https://stackoverflow.com/a/29378380
+static int ls_strtoi(const char *s, char **endptr, int base) {
+#if INT_MAX == LONG_MAX && INT_MIN == LONG_MIN
+  return (int)strtol(s, endptr, base);
+#else
+  return (int)ls_strtol_subrange(s, endptr, base, INT_MIN, INT_MAX);
+#endif
+} /* static int ls_strtoi */
+
+static int ls_init(void) {
+  livestatus_obj.socket_file = "/var/cache/naemon/live";
+  livestatus_obj.max_retry = 20;
+  livestatus_obj.backoff_sec = 1;
+
+  return 0;
+} /* int ls_init */
+
+static int ls_config(const char *key, const char *value) /* {{{ */
+{
+  if (strcasecmp("LivestatusSocketFile", key) == 0) {
+    char *socket_file = strdup(value);
+    if (socket_file == NULL) {
+      ERROR("livestatus plugin: strdup failed: %s", STRERRNO);
+      return -1;
+    }
+
+    livestatus_obj.socket_file = socket_file;
+  }
+
+  if (strcasecmp("OnFailureMaxRetry", key) == 0) {
+    livestatus_obj.max_retry = ls_strtoi(value, NULL, 10);
+    if (errno != 0) {
+      ERROR("livestatus plugin: strtoi failed for OnFailureMaxRetry: %s",
+            STRERRNO);
+      return -1;
+    }
+  }
+
+  if (strcasecmp("OnFailureBackOffSeconds", key) == 0) {
+    livestatus_obj.backoff_sec = ls_strtoi(value, NULL, 10);
+    if (errno != 0) {
+      ERROR("livestatus plugin: strtoi failed for OnFailureBackOffSeconds: %s",
+            STRERRNO);
+      return -1;
+    }
```

Comment:
A closing curly brace is missing here. This is causing the formatting issues. It will also lead to this code not compiling …

Reply:
I've recreated my dev environment, and obviously something was wrong here :) Thanks.